File: apps/workers-api/src/card-image-url.test.ts
import { describe, expect, it } from "vitest";
import { cardImageUrl } from "./card-image-url";

describe("cardImageUrl", () => {
  it("uses the canonical card image URL for every display context", () => {
    for (const variant of ["thumbnail", "list", "detail", "preview", "master"] as const) {
      expect(cardImageUrl("93", variant)).toBe("https://image.tcgcard.fun/cards/93.jpg");
    }
  });
});

File: apps/flutter-app/test/card_image_url_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:kando_app/shared/card_image/card_image_url.dart';

void main() {
  test('every display context uses the canonical card image URL', () {
    for (final variant in CardImageVariant.values) {
      expect(
        cardImageUrl('93', variant),
        'https://image.tcgcard.fun/cards/93.jpg',
      );
    }
  });
}
